SDWebImage/UIButton+WebCache.m

```diff
@@ -18,8 +18,14 @@
 
 typedef NSMutableDictionary<NSNumber *, NSURL *> SDStateImageURLDictionary;
 
+static inline NSNumber * backgroundImageURLKeyForState(UIControlState state) {
+    return @(NSUIntegerMax - state - 1);
```

Comment:
This looks like a hack. But for this usage(Just a map from state to URL) is OK. Maybe we could find a better solution like use prefix string as an key(`[NSString stringWithFormat:@"image_%@", @(state).stringValue]` and  `@"backgroundImage_%@"`)...